src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -505,6 +512,50 @@ private bool AgeCookies(string domain)
             return true;
         }
 
+        private void DomainTableCleanup()
+        {
+            var removePathList = new List<Object>();
+            var removeDomainList = new List<string>();
+
+            string currentDomain;
+            PathList pathList;
+
+            lock (m_domainTable.SyncRoot)
+            {
+                foreach (DictionaryEntry entry in m_domainTable)
```

Comment:
Unfortunately, using `foreach` with `Hashtable` like this results in a box allocation for each entry in the `Hashtable` because `Current` is typed as `object` so the `DictionaryEntry` struct has to be boxed/unboxed.

It'd be better to use the `IDictionaryEnumerator` directly, to avoid the unnecessary boxing, e.g.:

```c#
IDictionaryEnumerator enumerator = m_domainTable.GetEnumerator();
while (enumerator.MoveNext())
{
    currentDomain = (string)enumerator.Key;
    pathList = (PathList)enumerator.Value;
    ...
}
```

See #9050

Reply:
Remember to comment so that no one will "simplify" it back

Reply:
Incidentally I see 40 other cases already in corefx of foreach over a DictionaryEntry in the product sources. Don't know whether any of them matter enough to change.

Reply:
Will fix.